arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+2. Delete bucket from various cloud providers. Ex: Aws, gcp, azure
+
+3. Upload objects in bucket
+
+4. Set lifecycles rules on objects
+
+5. Migrate objects from one location to different locations
+
+
+##### Non Functional Requirements
+
+1. The system should be highly available. Multiple users can login to the system and create buckets, folders, upload objects etc.
+The design document for this can be referred from [here](https://github.com/sodafoundation/architecture-analysis/pull/96)
+
+2. The system should work with minimum latency.
+
+3. The system should be scalable and it should handle the high amount of users request.
+
+
+## Architecture Analysis
+
+The overall multi cloud architecture can be referred from [here](https://github.com/sodafoundation/architecture-analysis/blob/master/arch-design/multicloud/MultiCloud_Design.md)
+
+### Use case View
+
+For heterogeneous object management, bucket creation is important. Currently soda-multicloud supports creation of folders under cloud buckets as shown in Fig-1. So, soda-users must need to create cloud buckets manually and provide a name, AK/SK to soda for folder creation. As shown in fig-1, step-1 is manual which is performed by the user, whereas step-2,3,4,.. are supported by soda-multi-cloud.
+
+To avoid manual intervention, step-1 should also be part of soda-multicloud as shown in Fig-2. This way users need not to do anything manually on the cloud vendors console.
+
+Once the buckets are created, users should perform all the operations supported by soda-multicloud. For example: Uploading objects, setting lifecycle rules, migration etc.  
+
+
+![](resources/multi-cloud-fig1.png)
+
+
+![](resources/multi-cloud-fig2.png)
+
+
+
+#### API's(Create/Delete) to support bucket management:
+
+To support bucket managament, below is the list of API's with detailed description about parameters for various cloud vendors are below:
+
+
+#### AWS
+
+##### Rules for bucket naming:
+
+The following rules apply for naming S3 buckets:
+
+* Bucket names must be between 3 and 63 characters long.
+
+* Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
+
+* Bucket names must begin and end with a letter or number.
+
+* Bucket names must not be formatted as an IP address (for example, 192.168.5.4).
+
+* Bucket names can't begin with xn-- (for buckets created after February 2020).
+
+* Bucket names must be unique within a partition. A partition is a grouping of Regions. AWS currently has three partitions: aws (Standard Regions), aws-cn (China Regions), and aws-us-gov (AWS GovCloud [US] Regions).
+
+* Buckets used with Amazon S3 Transfer Acceleration can't have dots (.) in their names. For more information about transfer acceleration, see Amazon S3 Transfer Acceleration.
+
+
+##### URI Request Parameters
+
+The request uses the following URI parameters.
+
+`Bucket`: The name of the bucket to create.
+ Required: Yes
+
+`x-amz-acl`: The canned ACL to apply to the bucket.
```

Comment:
should all of these be vendor neutral ? no matter which back-end i am creating the buckets using SODA, my headers should be same. One of the inputs can be the back end specifier

Reply:
Yes. These SODA API's are vendor nuetral. From same API, user can create buckets on different clouds.